timesketch/lib/analyzers/feature_extraction.py

```diff
@@ -146,6 +178,54 @@ def run(self):
 
         return ', '.join(return_strings)
 
+    @staticmethod
+    def _get_attribute_value(
+            current_val,
+            extracted_value,
+            keep_multi,
+            merge_values,
+            type_list):
+        """Returns the attribute value as it should be stored.
+
+        Args:
+            current_val: current value of store_as.
+            extracted_value: values matched from regexp (type list).
+            keep_multi: choise if you keep all match from regex (type boolean).
+            merge_values: choise if you merge value from extracted
+                 and current (type boolean).
+            type_list: choise if you store values in list type(type boolean).
+
+        Returns:
+            Value to store
+        """
+        if not current_val:
+            merge_values = False
+        if len(extracted_value) == 1:
+            keep_multi = False
+        if type_list:
+            if merge_values and keep_multi:
+                return sorted(list(set(current_val) | set(extracted_value)))
+            if merge_values:
+                if extracted_value[0] not in current_val:
+                    current_val.append(extracted_value[0])
+                return sorted(current_val)
+            if keep_multi:
+                return sorted(extracted_value)
+            return [extracted_value[0]]
+        if merge_values and keep_multi:
+            extended_list = [x for x in extracted_value if not re.match(
+                f"(^|[^0-9a-zA-Z]){x}([^0-9a-zA-Z]|$)", current_val)
```

Comment:
I would opt to compile this RE

```
self._re_something_appropriate = re.compile(r'(......
```
and then here iuse

```
.... value if not self._re_whatever_name.match(current_val)
```

Reply:
Ok use split ',' instead of regexp